apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -205,7 +205,9 @@ type ClusterQueueStatus struct {
 	// +optional
 	// +listType=map
 	// +listMapKey=type
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Comment:
Why do we need this marker to add unit tests?

Reply:
Because the mock client uses `StrategicMerge` to implement SSA, which only looks at the patchStrategy, not the mapType.

There is an open issue to fix the mock client to be more like the real apiserver https://github.com/kubernetes/kubernetes/issues/115598

However, in the meantime, @apelisse suggested that I add a patchStrategy. This also gives the ability for external controllers to use client-side apply and easily merge two lists of conditions.

Reply:
Thanks for clarifying. I'm a bit confused since IIUC, the `StrategicMergePatch` can be used only for Client Side Apply (CSA), right?

However, by this PR, the controller uses SSA with `StarategicMergePatch` to update workload condition:

https://github.com/kubernetes-sigs/kueue/blob/483ef8a5c65d99fed60fef9eb0ce7809ef530199/pkg/workload/workload.go#L263

Which apply strategy will the controller use to update workload conditions? CSA vs SSA
Should we use a customized fake patch client to avoid performance issues instead of adding these markers if CSA is used?

https://github.com/kubernetes-sigs/controller-runtime/issues/2341#issuecomment-1689885336

Reply:
`Apply` means that it will use SSA.
It's only the fake client that implements SSA as it was CSA.

This is enough for our purposes for now.

Reply:
Ah, I see. So, CSA would be used only when unit tests.
Thanks again.